pymc3/plots/scatterplot.py

```diff
@@ -0,0 +1,136 @@
+import warnings
+
+try:
+    import matplotlib.pyplot as plt
+except ImportError:  # mpl is optional
+    pass
+import numpy as np
+from .utils import get_default_varnames
+from .artists import get_trace_dict, scale_text
+
+
+def scatterplot(trace, varnames=None, figsize=None, text_size=None, 
+                ax=None, hexbin=False, plot_transformed=False, divergences=False, 
+                kwargs_divergence=None, sub_varnames=None, **kwargs):
+
+    """
+    Plot a scatter or hexbin matrix of the sampled parameters.
+
+    Parameters
+    ----------
+
+    trace : result of MCMC run
+    varnames : list of variable names
+        Variables to be plotted, if None all variable are plotted
+    figsize : figure size tuple
+        If None, size is (8 + numvars, 8 + numvars)
+    text_size: int
+        Text size for labels
+    ax : axes
+        Matplotlib axes.
+    hexbin : Boolean
+        If True draws an hexbin plot
+    plot_transformed : bool
+        Flag for plotting automatically transformed variables in addition to
+        original variables (defaults to False). Applies when varnames/sub_varnames = None.
+        When a list of varnames/sub_varnames is passed, transformed variables can be passed
+        using their names.
+    divergences : Boolean
+        If True divergences will be plotted in a diferent color
+    kwargs_divergence : dicts, optional
+        Aditional keywords passed to ax.scatter for divergences
+    sub_varnames : list, optional        
+        Aditional varnames passed for plotting subsets of multidimensional variables
+    Returns
+    -------
+
+    ax : matplotlib axes
+
+    """    
+    
+    if varnames is None:
+        if plot_transformed:
+            trace_dict = get_trace_dict(trace, get_default_varnames(trace.varnames, True))
+        else:
+            trace_dict = get_trace_dict(trace, get_default_varnames(trace.varnames, False))
+        if sub_varnames is None:
+            varnames = list(trace_dict.keys())
+        else:
+            varnames = sub_varnames
+    else:
+        trace_dict = get_trace_dict(trace, varnames)
+        varnames = list(trace_dict.keys())       
+
+    if text_size is None:
+        text_size = scale_text(figsize, text_size=text_size)
+
+    if kwargs_divergence is None:
+        kwargs_divergence = {}
+    
+    numvars = len(varnames)
+
+    if figsize is None:
+        figsize = (8+numvars, 8+numvars)    
+
+    if numvars < 2:
+        raise Exception('Number of variables to be plotted must be 2 or greater.')
+        
+    if numvars == 2 and ax is not None:
+        if hexbin:
+            ax.hexbin(trace_dict[varnames[0]], trace_dict[varnames[1]], mincnt=1, **kwargs)
+        else:
+            ax.scatter(trace_dict[varnames[0]], trace_dict[varnames[1]], **kwargs)
+        
+        if divergences:
+            try:
+                divergent = trace['diverging']
```

Comment:
It would be cleaner to use `trace.get_sampler_stats`, just in case there is a var named 'diverging'.